frontstage/views/sign_in/sign_in.py

```diff
@@ -67,10 +67,25 @@ def login():
             return render_template('sign-in/sign-in.html', form=form, data={"error": {"type": "failed"}})
         party_id = party_json['id']
 
-        oauth2_token = get_oauth_token(username, password, party_id, party_json, form, request.args.get('next'))
-
-        if not isinstance(oauth2_token, dict):
-            return oauth2_token
+        try:
+            oauth2_token = oauth_controller.sign_in(username, password)
+        except OAuth2Error as exc:
+            error_message = exc.oauth2_error
+            if USER_ACCOUNT_LOCKED in error_message:
+                logger.info('User account is locked on the OAuth2 server', party_id=party_id)
```

Comment:
Could we add the status (from `party_json`) to this log message? May aid debugging.